src/pytorch_lightning/core/mixins/device_dtype_mixin.py

```diff
@@ -116,14 +116,16 @@ def cuda(self, device: Optional[Union[torch.device, int]] = None) -> Self:  # ty
         while being optimized.
 
         Arguments:
-            device: if specified, all parameters will be
-                copied to that device
+            device: If specified, all parameters will be copied to that device. If `None`, the current CUDA device
+                index will be used.
 
         Returns:
             Module: self
         """
-        if device is None or isinstance(device, int):
-            device = torch.device("cuda", index=(device or 0))
```

Comment:
Why not revert to the old version?

`torch.device("cuda", index=device)`

It supports None and int and puts the choice on torch.

If mypy is not happy then we ignore it. This shouldnt have been changed

Reply:
I couldn't find documentation on torch for this behavior. It is better to have the logic there in case undocumented behavior changes.